util/telemetry/instrument.go

```diff
@@ -155,3 +155,23 @@ func (m *Metrics) buckets(name string, defaultBuckets []float64) []float64 {
 	}
 	return defaultBuckets
 }
+
+func (i *Instrument) GetName() string {
+	return i.name
+}
+
+func (i *Instrument) GetDescription() string {
+	return i.description
+}
+
+func (i *Instrument) GetOtel() interface{} {
+	return i.otel
+}
+
+func (i *Instrument) SetUserdata(data interface{}) {
+	i.userdata = data
+}
+
+func (i *Instrument) GetUserdata() interface{} {
+	return i.userdata
+}
```

Comment:
Why not export these fields? 
Sure someone can mutate them, but userdata can be set, otel is an interface.

No big issues though, just a suggestion, feel free to ignore.

Reply:
Mutating the fields which are read-only would cause pretty bad damage internally. Previously these were all internal to the workflow/`metrics` package, but Userdata needs mangling by custom metrics, so I had to expose it as RW. At least it's explicit.